### ISSUE TITLE
ContextErrorException Warning: Illegal offset type in isset or empty

### ISSUE THREAD

**Issue by Aerendir** [VincentChalnot/SidusEAVModelBundle#17]
When creating the services for families, this error happens.

The problem is the way the services for families are created: the use of the `Parameter` class is the problem.

Fetching the class name directly from the container fixes the problem:

       $definition = new Definition(
            $container->getParameter('sidus_eav_model.family.class'),
            [
                $code,
                new Reference('sidus_eav_model.attribute.registry'),
                new Reference('sidus_eav_model.family.registry'),
                new Reference('sidus_eav_model.context.manager'),
                $familyConfiguration,
            ]
        );

**Comment by VincentChalnot** [comment]
Hello ! Thanks for your contribution to this project :v: 
Unfortunately I can't reproduce this issue.
What version of Symfony and what version of this bundle are you using ?
I recommend Symfony >3.1 with sidus/eav-model-bundle: v1.2.*.
Can you give me a config example ?

**Comment by Aerendir** [comment]
Hello @VincentChalnot , glad to help :)

I'm using Symfony `3.3.*` and EavModelBundle `~1.2`.

I did nothing special: simply started to follow the tutorial.

I set the a simple configuration:

```
sidus_eav_model:
    data_class: AppBundle\Entity\EavData
    value_class: AppBundle\Entity\EavValue
    attributes:
        title:
            type: string
            group: this_is_the_group
            options:
                allowed_families: product
                hidden: false
    families:
        product:
            attributes:
                title: ~
```

Nothing special as you can see...

**Comment by FabienSalles** [comment]
Hello @VincentChalnot ,

This error occurs when we have a service using the autowiring feature.

Symfony try to use the class in this line of code ``` if (isset($this->classReflectors[$class])) {``` (in Symfony/Component/DependencyInjection/ContainerBuilder.php:344) but we have a Parameter object : 
```
class Symfony\Component\DependencyInjection\Parameter#5708 (1) {
  private $id =>
  string(28) "sidus_eav_model.family.class"
}
```
instead of just the class name  : ```Sidus\EAVModelBundle\Model\Family```

When we look to the code of the Definition object, the class parameter have to be a string or null, not a Parameter object : 
```
/**
     * @param string|null $class     The service class
     * @param array       $arguments An array of arguments to pass to the service constructor
     */
    public function __construct($class = null, array $arguments = array())
    {
        if (null !== $class) {
            $this->setClass($class);
        }
        $this->arguments = $arguments;
    }
``` 

So I created a PR #24 :)

**Comment by VincentChalnot** [comment]
Resolved in PR #24 !
Thanks guys
